pkg/clustermesh/types/option.go

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package types
+
+import (
+	"github.com/spf13/pflag"
+
+	"github.com/cilium/cilium/pkg/defaults"
+)
+
+const (
+	// OptClusterName is the name of the OptClusterName option
+	OptClusterName = "cluster-name"
+
+	// OptClusterID is the name of the OptClusterID option
+	OptClusterID = "cluster-id"
+)
+
+// ClusterIDName groups together the ClusterID and the ClusterName
+type ClusterIDName struct {
```

Comment:
nit: No keen on the naming here for 3 reasons:

1. Naming the struct for what it contains vs what it provides i.e ClusterInfo or ClusterMeta.
2. We are already in the context of a cluster thus we could dry the members names aka ID, Name if we weren't using anonymous aggregation that is...
3. If we were to add more fields later on the current name would be a bit of a dud.

Reply:
Yeah, I agree that this naming was suboptimal. For context, when I introduced this `ClusterIDName` struct a while ago, it was essentially meant to be embedded inside other structs, which is the reason for the more verbose `ClusterID`/`ClusterName` field names. I've now renamed the struct to `ClusterInfo`, and the fields to `ID` and `Name`.

Reply:
So... This change ended up breaking everything, because the name of the flag must match that of the target struct field. Let me rework this differently.

Reply:
It turned out to be easier than expected, just needed to add the tags.